src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const array = [];
+  let oldState = { ...state };
+
+  for (const action of actions) {
+    const newState = { ...oldState };
+
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(newState, action.extraData);
+        array.push(newState);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete newState[key];
+        }
+        array.push(newState);
+        break;
+
+      case 'clear':
+        for (const key in newState) {
+          delete newState[key];
+        }
+        array.push(newState);
+        break;
+
+      default:
+        // do nothing for unknown action types
```

Comment:
remove comments, better to not write them or write if it is something super-tricky to understand(but in more cases if it's tricky - it can be rewritten simplier :)

In your case it's just not needed